firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -0,0 +1,163 @@
+   * update is found, returns null. If tester is not signed in, presents the tester with a Google
+   * sign in UI
+   */
+  @NonNull
+  public Task<AppDistributionRelease> checkForUpdate() {
+    return Tasks.forResult(null);
+  }
+
+  /**
+   * Updates app to the latest release. If the latest release is an APK, downloads the binary and
+   * starts an installation If the latest release is an AAB, directs the tester to the Play app to
+   * complete the download and installation.
+   *
+   * @throws an UPDATE_NOT_AVAIALBLE exception if no new release is cached from checkForUpdate
+   * @param updateProgressListener a callback function invoked as the update progresses
+   */
+  @NonNull
+  public Task<Void> updateApp(@Nullable UpdateProgressListener updateProgressListener) {
+    return Tasks.forResult(null);
+  }
+
+  /**
+   * Interface to subscribe to status updates on the release update process. Called by updateApp.
+   */
+  public interface UpdateProgressListener {
+    public void onProgressUpdate(@NonNull UpdateProgress updateProgress);
+  }
+
+  public interface UpdateProgress {
+    // The number of bytes downloaded so far for the APK.
+    //  Returns -1 if called on an AAB.
+    @NonNull
+    public long getApkBytesDownloaded();
+
+    // The total number of bytes to download for the APK.
+    // Returns -1 if called on an AAB.
+    @NonNull
+    public long getApkTotalBytesToDownload();
+
+    @NonNull
+    public UpdateStatus getUpdateStatus();
+  }
+
+  public enum UpdateStatus {
+    // Update queued but not started
+    PENDING,
+
+    // Download in progress
+    DOWNLOADING,
+
+    // Download completed
+    DOWNLOADED,
+
+    // Download failed
+    DOWNLOAD_FAILED,
+
+    // Update installed
+    INSTALLED,
+
+    // Installation cancelled
+    INSTALL_CANCELED,
+
+    // Installation failed
+    INSTALL_FAILED,
+
+    // AAB flow (directed to Play)
+    REDIRECTED_TO_PLAY,
+  }
+
+  /** Signs in the App Distribution tester. Presents the tester with a Google sign in UI */
+  @NonNull
+  public Task<Void> signInTester() {
+    return Tasks.forResult(null);
+  }
+
+  /** Returns true if the App Distribution tester is signed in */
+  @NonNull
+  public boolean isTesterSignedIn() {
+    return false;
+  }
+
+  /** Signs out the App Distribution tester */
+  public void signOutTester() {}
+
+  /** The release information returned by the update check when a new version is available. */
+  public interface AppDistributionRelease {
+    // The short bundle version of this build (example 1.0.0)
+    @NonNull
+    public String getDisplayVersion();
+
+    // The bundle version of this build (example: 123)
+    @NonNull
+    public String getBuildVersion();
+
+    // The release notes for this build
+    @NonNull
+    public String getReleaseNotes();
+
+    // The binary type for this build
+    @NonNull
+    public BinaryType getBinaryType();
+  }
+
```

Comment:
move this outside of the FirebaseAppDistribution class (not sure if this should be in its own file). @vkryachko thoughts?